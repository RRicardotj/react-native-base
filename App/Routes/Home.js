import React from 'react';

import {
  Button,
  StyleSheet,
  Text,
  View,
} from 'react-native';

import Styles from '../Utils/Styles';
import Constants from '../Utils/Constants';
import Routes from '../Navigation/Routes';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  title: {
    fontSize: 32,
    fontWeight: 'bold',
    color: 'white',
    textAlign: 'center',
  },
  desc: {
    marginVertical: 5,
    fontSize: 18,
    color: 'white',
    textAlign: 'center',
  },
});

export default class Home extends React.Component {
  static propTypes = {
    navigator: React.PropTypes.object.isRequired,
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{Constants.APP_NAME}</Text>
        <Text style={styles.desc}>{Constants.APP_DESC}</Text>

        <Button
          text="About"
          color="#841584"
          onPress={() => this.props.navigator.push(Routes.about())}
          isFilled />
      </View>
    );
  }
}
