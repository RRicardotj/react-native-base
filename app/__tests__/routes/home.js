import 'react-native';
import React from 'react'; // eslint-disable-line no-unused-vars
import { shallow } from 'enzyme';

import Home from '../../routes/home';

import renderer from 'react-test-renderer';

describe('routes/home.js', () => {
  const props = {
    navigation: {
      push: jest.fn()
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <Home {...props} />
    );

    expect(tree).toMatchSnapshot();
  });

  it('should navigate to the info view', () => {
    const wrapper = shallow(
      <Home {...props} />
    );

    expect(wrapper).toBeDefined();

    wrapper.find('Button').simulate('press');
    expect(props.navigation.navigate).toHaveBeenCalled();
  });
});
