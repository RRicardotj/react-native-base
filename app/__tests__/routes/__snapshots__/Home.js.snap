// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/home.js renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 26,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    React Native Base
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "marginVertical": 5,
        "textAlign": "center",
      }
    }
  >
    Add-On for React Native
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#2359BA",
            },
          ]
        }
      >
        More Information
      </Text>
    </View>
  </View>
</View>
`;
